### PR TITLE
feat: add directed graph and supported modules

### LINKED ISSUE #5: Add directed graph and support modules
## 🚀 Feature

Added a directed graph, which was agreed to be used to process and return information about how the cells which were detected inside the table region span and the supported modules.

## Motivation

As agreed, a directed graph will be used to process and return information about how the cells which were detected inside the table region span.  This information is very important because it determines whether the table structure after being refactored is as expected

## Pitch
A new module containing directed graphs and auxiliary modules should be added to the source code, and the code that uses them to process the results of table detection and line segmentation should also be added.

The new module should be well organized, type hinting, and fully docstring added.

## Alternatives

None

## Additional context

None

### PR DESCRIPTION
**Pull request by manhbq-2536**
## Description

Add directed graph and supported modules

Closes #5 

## Type of change

Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Add more tests
- [ ] Others
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Create and activate the virtual environment
- Install package by running python setup.py install in the root folder
- Take the output of line segmentation and return XY coordinate of cells

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have successfully run the modified code in my local environment with no errors
- [ ] I ran the proper linters and fixed the issues pointed out to make sense
- [ ] I updated the unit test based on my edit

### CODE REVIEW

**Comment by trungtd-2436** [comment]
@manhbq-2536 squash lại commit nhé Mạnh

File: table_reconstruction/utils/cell_utils.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
+from shapely.geometry import LineString
+from table_reconstruction.utils.lines_utils import is_line
+import numpy as np
+
+
+def get_intersection_points(horizontal_lines, vertical_lines, tab_coord):
```
Comment (chienlm-2537) [inline table_reconstruction/utils/cell_utils.py:6]: thiếu type hint nhé

File: table_reconstruction/utils/cell_utils.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    if np.all(nearest_point_metric > ths):
+        return None, None
+    else:
+        index = np.argmin(nearest_point_metric)
+        bottom_right_vertices = points[index]
+
+        return (bottom_right_vertices[0], bottom_right_vertices[1])
+
+
+def calculate_cell_coordinate(points, fake_flag, ths, *lines):
```
Comment (chienlm-2537) [inline table_reconstruction/utils/cell_utils.py:117]: Thiếu type hint

File: table_reconstruction/utils/lines_utils.py [hunk 3]
```diff
@@ -0,0 +1,507 @@
+from skimage import measure
+from scipy.spatial import distance as dist
+import numpy as np
+from table_reconstruction.utils import mask_utils
+import cv2
+
+
+def get_table_line(binimg, axis, lineW=5):
```
Comment (chienlm-2537) [inline table_reconstruction/utils/lines_utils.py:8]: Thiếu type hint nhé :v

Comment (manhbq-2536) [reply table_reconstruction/utils/lines_utils.py]: Done nhé anh ơi

File: table_reconstruction/utils/cell_utils.py (hunk 2 above)
Comment (manhbq-2536) [reply table_reconstruction/utils/cell_utils.py]: Done nhé anh ơi

File: table_reconstruction/utils/cell_utils.py (hunk 1 above)
Comment (manhbq-2536) [reply table_reconstruction/utils/cell_utils.py]: Done nhé anh ơi

**Pull request closed without merging**